src/agent/Makefile

```diff
@@ -105,8 +105,10 @@ default: $(TARGET) show-header
 
 $(TARGET): $(GENERATED_CODE) $(TARGET_PATH)
 
+#remember to add '-C link-arg=-lgcc' to RUSTFLAGS when you assign it.
 $(TARGET_PATH): $(SOURCES) | show-summary
-	@RUSTFLAGS="--deny warnings" cargo build --target $(TRIPLE) --$(BUILD_TYPE)
+	@RUSTFLAGS='-C link-arg=-lgcc --deny warnings' cargo build --target $(TRIPLE) --$(BUILD_TYPE)
```

Comment:
IIUC this is only needed for `aarch64`. Could you make it `aarch64` specific? Not sure how to switch in Makefile though. @jodh-intel (aka, our lord of Makefiles) do you have any suggestions?